feature-libs/product-configurator/rulebased/core/state/effects/configurator-cart.effect.ts

```diff
@@ -85,19 +97,21 @@ export class ConfiguratorCartEffects {
               return [
                 new CartActions.CartUpdateEntrySuccess({
                   ...entry,
-                  userId: payload.userId,
+                  userId: payload.userId || '',
                   cartId: payload.cartId,
-                  entryNumber: entry.entry.entryNumber.toString(),
-                  quantity: entry.quantity,
+                  entryNumber: entry.entry?.entryNumber
+                    ? entry.entry.entryNumber.toString()
+                    : '0',
+                  quantity: entry?.quantity || 1,
                 }),
               ];
             }),
             catchError((error) =>
               of(
                 new CartActions.CartUpdateEntryFail({
-                  userId: payload.userId,
+                  userId: payload.userId || '',
                   cartId: payload.cartId,
-                  entryNumber: payload.cartEntryNumber,
+                  entryNumber: payload.cartEntryNumber || '0',
```

Comment:
1. Are parameters userId and cartEntryNumber _really_ optional in the occ call?
```
  export interface UpdateConfigurationForCartEntryParameters {
    userId?: string;
    cartId: string;
    cartEntryNumber?: string;
    configuration: Configurator.Configuration;
  }
```
If not, we can make them required -> then we don't need some fallbacks here.

2. What about the repsonse? Should we throw an error if we're missing `entry`? Should we really fallback to '0' or 1?

Reply:
When I read CartEntryEffects correctly, quantity and entryNumber are not needed. So I propose to at least have quantity as optional, and for now keep entryNumber as mandatory (not technically needed here, but essential for the update process)